src/lib/viewers/BaseViewer.js

```diff
@@ -1259,12 +1249,21 @@ class BaseViewer extends EventEmitter {
         // we remain in create mode
         if (status === 'success') {
             this.annotator.emit('annotations_active_set', id);
+
+            if (this.annotationControls) {
+                this.annotationControls.resetTempMode();
+            }
         }
     }
 
-    handleAnnotationModeChangeEvent({ mode }) {
-        if (this.annotationControls) {
-            this.annotationControls.setMode(mode);
+    handleAnnotationStagedChangeEvent({ status, type }) {
+        if (!this.annotationControls) {
+            return;
+        }
+        if (!status || status === 'cancel') {
+            this.annotationControls.resetTempMode();
```

Comment:
Should the controls component no longer be managing the "current mode" state at all? Only which (if any) control is active?